src/goto-programs/goto_trace.cpp

```diff
@@ -75,7 +75,11 @@ void goto_trace_stept::output(
   case goto_trace_stept::SHARED_WRITE: out << "SHARED WRITE"; break;
   case goto_trace_stept::FUNCTION_CALL: out << "FUNCTION CALL"; break;
   case goto_trace_stept::FUNCTION_RETURN: out << "FUNCTION RETURN"; break;
-  default: assert(false);
+  case goto_trace_stept::DEAD: out << "DEAD" << std::endl; return;
+  default: {
```

Comment:
My suggestion would be to omit those default cases - the compiler will warn about missing cases.